pkg/scheduler/server/api.go

```diff
@@ -61,111 +66,93 @@ func (s *Server) ScheduleJob(ctx context.Context, req *schedulerv1pb.ScheduleJob
 	return &schedulerv1pb.ScheduleJobResponse{}, nil
 }
 
-func (s *Server) triggerJob(job *runtimev1pb.Job, namespace string, metadata map[string]string) error {
-	_, err := s.TriggerJob(context.Background(), &schedulerv1pb.TriggerJobRequest{
-		JobName:   job.GetName(),
-		Namespace: namespace,
-		Metadata:  metadata,
-	})
-	if err != nil {
-		log.Errorf("error triggering job %s: %s", job.GetName(), err)
-		return err
-	}
-	return nil
-}
+func (s *Server) triggerJob(ctx context.Context, metadata map[string]string, payload *anypb.Any) (etcdcron.TriggerResult, error) {
+	log.Debug("Triggering job")
+	actorType := metadata["actorType"]
+	actorID := metadata["actorId"]
+	reminderName := metadata["reminder"]
+	if actorType != "" && actorID != "" && reminderName != "" {
+		if s.actorRuntime == nil {
+			return etcdcron.Failure, fmt.Errorf("actor runtime is not configured")
+		}
 
-// ListJobs is a placeholder method that needs to be implemented
-func (s *Server) ListJobs(ctx context.Context, req *schedulerv1pb.ListJobsRequest) (*schedulerv1pb.ListJobsResponse, error) {
-	select {
-	case <-ctx.Done():
-		return nil, ctx.Err()
-	case <-s.readyCh:
-	}
+		invokeMethod := "remind/" + reminderName
+		contentType := metadata["content-type"]
+		invokeReq := internalv1pb.NewInternalInvokeRequest(invokeMethod).
+			WithActor(actorType, actorID).
+			WithData(payload.GetValue()).
+			WithContentType(contentType)
 
-	entries := s.cron.ListJobsByPrefix(req.GetAppId() + "||")
-
-	jobs := make([]*runtimev1pb.Job, 0, len(entries))
-	for _, entry := range entries {
-		job := &runtimev1pb.Job{
-			Name:     entry.Name,
-			Schedule: entry.Rhythm,
-			Repeats:  entry.Repeats,
-			DueTime:  entry.DueTime,
-			Ttl:      entry.TTL,
-			Data:     entry.Data,
+		res, err := s.actorRuntime.Call(ctx, invokeReq)
+		if err != nil {
+			return etcdcron.Failure, err
 		}
 
-		jobs = append(jobs, job)
-	}
+		if res.GetStatus().GetCode() != int32(codes.OK) {
+			return etcdcron.Failure, nil
+		}
 
-	resp := &schedulerv1pb.ListJobsResponse{Jobs: jobs}
+		return etcdcron.OK, err
+	}
 
-	return resp, nil
+	// Trigger not supported
+	log.Warn("Cannot trigger job: not supported.")
+	return etcdcron.Failure, nil
```

Comment:
Why do we not return an error here? We need at least the name of the job in the log line.

Reply:
This seems to happen when the metadata does not have all the correct fields. Please move this higher in a if block which checks if any of the fields are empty.

Reply:
There will be more conditions here, so combining multiple ones early can be bug-prone and duplicate the logic on future "if" conditions.

Reply:
Added metadata to the output since callback does not have jobName (yet).